### DISCUSSION TITLE
Why are excluded files still analyzed?

### DISCUSSION THREAD

**Discussion by jpenna** [golangci/golangci-lint#4966] [Q&A] [answered]
If the files are excluded from reporting, why are they analyzed? 

Even if they are imported by other files, the static analysis required for them is much less thorough them the "importer file"...

As a concrete example, there were more than 2k issues that were found but not reported:

```
 level=info msg="[runner/skip_dirs] Skipped 1949 issues from dir db/client by pattern db"
  level=info msg="[runner/skip_dirs] Skipped 131 issues from dir proto/generated by pattern proto/generated"
  level=info msg="[runner/skip_dirs] Skipped 280 issues from dir db/models by pattern db"
  level=info msg="[runner/skip_dirs] Skipped 4 issues from dir db by pattern db"
```

**Comment by ldez** [comment, marked as answer]
Hello,

it's how the Go tooling works, we have real solution for now: https://github.com/golangci/golangci-lint/issues/4360#issuecomment-2066717664
